File: TripKitAndroid/src/main/java/com/skedgo/geocoding/agregator/GCAppResultInterface.java
package com.skedgo.geocoding.agregator;

/**
 * Information that the user saves in the app
 */
public interface GCAppResultInterface extends GCResultInterface {

    //  address value
    String getSubtitle();

    //  source that provides the result
    Source getAppResultSource();

    //    is true if the result was set as favourite by the user false otherwise.
    boolean isFavourite();

    //   source that provides the result
    enum Source {
        AddressBook,
        Regions,
        Calendar,
        History
    }

}

File: TripKitAndroid/src/main/java/com/skedgo/geocoding/agregator/GCAppResultInterface.kt
package com.skedgo.geocoding.agregator

/**
 * Information that the user saves in the app
 */
interface GCAppResultInterface : GCResultInterface {

    // Address value
    val subtitle: String

    // Source that provides the result
    val appResultSource: Source

    // True if the result was set as favorite by the user, false otherwise
    val isFavourite: Boolean

    // Source that provides the result
    enum class Source {
        AddressBook,
        Regions,
        Calendar,
        History
    }
}
